server/src/main/java/org/elasticsearch/script/CtxMap.java

```diff
@@ -150,7 +151,7 @@ public Object remove(Object key) {
     @Override
     public void clear() {
         // AbstractMap uses entrySet().clear(), it should be quicker to run through the validators, then call the wrapped maps clear
-        for (String key : metadata.keySet()) {
+        for (String key : new ArrayList<>(metadata.keySet())) { // copy the key set to get around the ConcurrentModificationException
             metadata.remove(key);
         }
         // TODO: this is just bogus, there isn't any case where metadata won't trip a failure above?
```

Comment:
We could remove this TODO now

Reply:
Ehhhhhhh, I rewrote the comment in ce216ff06d5b098cb2608b68baee83c6e65e6946, but I didn't remove it. `clear()` still doesn't work in the general case, for sure, and I'm not sure it works in any particularly practical case.